### ISSUE TITLE
Angel's Petrochem compatybility

### ISSUE THREAD

**Issue by SQLek** [TheRealGangsir/SupplyDrop#1]
Angels petrochem shadows vanilia oil research and subsitude its own in place.

I made a patch that check both research and also don't crash if there is no angels research.

So far Your mod is Angel's compatybile and Bob's compatybile, no idea if other big changing mods like Yuoki do similar trick like Angel petrochem

[angels_petrochem_compatybility.patch.txt](https://github.com/TheRealGangsir/SupplyDrop/files/818906/angels_petrochem_compatybility.patch.txt)

Ugh. I have to change extension to txt.

PS Are there another options for contacting You other than redit?

**Comment by Sublimelime** [comment]
@SQLek Yes, if you're a part of the factorio discord, you can also find me there.

The aim of my mod is to be balanced to vanilla, so I'm not interested in adding special compatibility for inter-mod checking. This may change in the future, but until then, glad you enjoy my mod and thanks for the suggestion.

**Comment by SQLek** [comment]
Still broken in 1.2

> The aim of my mod is to be balanced to vanilla

This patch don't change any drop chances or drop slots. I spend all day testing for regressions.

I know You hesitate to apply patch from stranger right away, but to lazy to read?

`local isOilProcessingResearched = (randomPlayer.force.technologies["angels-oil-processing"]
and randomPlayer.force.technologies["angels-oil-processing"].researched
or randomPlayer.force.technologies["oil-processing"].researched)`

`local isAdvancedOilProcessingResearched = (randomPlayer.force.technologies["angels-advanced-oil-processing"]
and randomPlayer.force.technologies["angels-advanced-oil-processing"].researched
or randomPlayer.force.technologies["advanced-oil-processing"].researched)`

And in conditions it uses **isOilProcessingResearched** and **isAdvancedOilProcessingResearched**

Give me a reason that You don't applied my patch, that i extensively tested? It doesn't change balance of drops, it works with vanilla, it gives You ability to support other mods too without very long lines. Why?

**Comment by Sublimelime** [comment]
@SQLek Ah, I see what you're trying to get me to do. Sure, I'd be willing to fix it. Could you instead of giving me a patch file, submit a pull request? Makes it a bit easier for me to see exactly what changes were made, and lets me merge it automatically. :)

**Comment by SQLek** [comment]
Thank You. I'll learn how to use pull functionality.

And sorry if my language was harsh. I'm not English native.

Stay cool.
